WCコマンドを作成

sochi419 opened the pull request:
wcコマンドを作成しました。
確認よろしくお願いします🙏

sochi419 committed:
first commit

sochi419 committed:
change output side by side

sochi419 commented:
出力が縦並びだったものを、横並びにしました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+
+def main
+  opt = ARGV.getopts('l', 'w', 'c')
+  if ARGV.size.zero?
```
```suggestion
  if ARGV.empty?
```

でもよさそうです

sochi419 replied:
`empty`を使った記法に修正しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+def main
+  opt = ARGV.getopts('l', 'w', 'c')
+  if ARGV.size.zero?
+    use_stdin(opt) # コマンドに引数が与えられなかった場合。
```
説明的なコメントは基本的になしでOKです。

参考 [【動画付き・初心者向け】コメントにはWHYを書こう 〜不要なコメントと必要なコメントの違いについて〜 \- Qiita](https://qiita.com/jnchito/items/f0d90af4ed44b7484103)

あと、`use_stdin`だと「標準入力を使うメソッド」という意味ですが、メソッド役割としては「標準入力や起動時引数を使って○○する」という方が主になるはずなので、メソッド名はそっちに寄せた方がいいと思います。

sochi419 replied:
不要だと感じた説明コメントは削除しました。

`use_stdin`を`output_stdin`
`use_argv`を`output_argv`
に命名変更しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+    print word_number.to_s.rjust(8, ' ') # 単語数
+    print bytesize.to_s.rjust(8, ' ') # バイト数
+  else
+    output_stdin_option(opt, inputs, word_number, bytesize)
```
うまくやればこのメソッドに出力処理を一本化できそうです。
なるべくDRYなコードを目指しましょう。

sochi419 replied:
出力処理を1本化しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+end
+
+def output_stdin_option(opt, inputs, word_number, bytesize)
+  print inputs.size.to_s.rjust(8, ' ') if opt['l'] == true
```
`to_s.rjust(8, ' ')`が何度も出てくるのでDRYにしましょう

あと、`== true`や`== false`は通常書かないです。改訂版チェリー本を持ってたら2章のコラムを読んでみてください〜。
<img width="624" alt="Screen Shot 2022-08-28 at 15 29 46" src="https://user-images.githubusercontent.com/1148320/187060917-eef999e2-679e-455d-b252-8fab2e0b0903.png">

sochi419 replied:
`to_s.rjust(8, ' ')`が何度も出てくるので、メソッドとして切り分けました。
````
def output(result)
  print result.to_s.rjust(8, ' ')
end
````

また、`== true`や`==false`を使わない記法に変更しました。
````
if (opt['l'] == false && opt['w'] == false && opt['c'] == false) || (opt['l'] == true && opt['w'] == true && opt['c'] == true) # 修正前
````
````
if (!opt['l'] && !opt['w'] && !opt['c']) || (opt['l'] && opt['w'] && opt['c']) # 修正後
````

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+# コマンドに引数が与えられなかった場合。
+def use_stdin(opt)
+  inputs = $stdin.readlines
+  word_number = 0
```
word_numberだと「番号」っぽいので、word_countとかの方がいいかもしれません。

sochi419 replied:
`word_count`に修正しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+  total_lines_words_bytes = []
+  n = 0
+
+  until n == (ARGV.size) # コマンドライン引数の数だけ、ループをさせる。
```
untilでループさせるより、eachメソッド等を活用した方がRubyらしいです。

```ruby
ARGV.each do |hoge|
```

あと、ARGVは特殊な定数なので、あちこちのメソッドで依存させるとメソッドの再利用性が悪くなります。
ARGVが登場するのはmainメソッドだけにして、あとはメソッドの引数で渡していくのがお勧めです。

sochi419 replied:
untilのループ文を、eachを使ったループ文に変更しました。
また、ARGVはmainメソッドのみで使用し、メソッドで使用するときに引数で渡す処理に変更しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+end
+
+def count_words(file)
+  ary = file.split(/\s+/)
```
20行目と微妙にカウントのロジックが異なりますね。
共通化してしないと同じテキストなのに入力方法によってカウントが異なる、といった問題が起きるかもしれません。

sochi419 replied:
カウントのロジックを統一しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+
+def count_words(file)
+  ary = file.split(/\s+/)
+  ary.size
```
変数に入れるほどの処理ではないかも？

```ruby
file.split(/\s+/).size
```

sochi419 replied:
変数に入れる処理を消しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+  print 'total'.ljust(count_max_word(ARGV), ' ')
+end
+
+def output_argv(opt, argument, _output)
```
```suggestion
def output_argv(opt, argument)
```

引数は2つだけでいいのでは？

sochi419 replied:
3つ目の引数は仰る通り、不要でした。
引数を2つに変更しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+
+# コマンドに引数が与えられた場合。
+def use_argv(opt)
+  lines_words_bytes = []
```
この変数、使ってます？

sochi419 replied:
すみません、この変数も不要でした。
削除しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+  end
+end
+
+def output_argv_option(opt, str, argument)
```
これも36行目のメソッドとよく似てるのでDRYにできそうな気がします

sochi419 replied:
36行目同様、出力処理を1本化しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,107 @@
+    totals = total_lines_words_bytes.transpose.map(&:sum)
+
+    print(' ')
+    print ARGV[n].to_s.ljust(count_max_word(ARGV), ' ')
```
ファイル名は最後尾の列なので、

```
a.txt___
abcd.txt
```

のように無駄にスペースを入れる必要はなく、


```
a.txt
abcd.txt
```

と素直に出力すれば良いと思いますー

sochi419 replied:
無駄なスペースを無くした処理に変更しました。

JunichiIto reviewed (commented):
コメントしました！

ところで、データを作る処理とprintする処理を明確に分けると、見通しがよくDRYなコードが書きやすくなります。
イメージ的にはこんな感じです（あくまでイメージです。メソッド名や変数名、引数の設計等は適当なので、そのまま使わないように！）

```ruby
# 標準入力でもファイル入力でも、まったく同じデータ構造のデータが返る
data =
  if stdin?
    build_data_for_stdin
  else
    build_data_for_files(ARGV)
  end

# putsやprintを行うのはこのメソッドだけ
print_data(data)
```

一度この方向でコードが修正できないか、ちょっと検討してみてください〜。

sochi419 committed:
The main changes are:
・The process of creating and outputting data was separated.
・Fixed variable names and method names

sochi419 commented:
レビューありがとうございます！
修正しました！

主な変更は以下です。
・データを作る処理と出力をする処理を分けた。
・変数名、メソッド名の修正をした
その他細かい箇所の修正は、PRのコメントに返信しています。

以上、確認よろしくお願いします🙏

sochi419 committed:
Fixed the process that % is not displayed in the output

sochi419 commented:
出力結果に%が付随してしまっていたので、修正しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,104 @@
+  opt = ARGV.getopts('l', 'w', 'c')
+
+  if ARGV.empty?
+    datas = build_data_for_stdin
```
dataはdatumの複数形なので、datasは英文法的にNGです。
[モデルやメソッドに名前を付けるときは英語の品詞に気をつけよう \- Qiita](https://qiita.com/jnchito/items/459d58ba652bf4763820)

そもそも`data`という変数名は「プログラミング中に出てくる変数なんて、みんな何らかのデータやん」という話になってしまうので、もうちょっと具体的な名前を付けた方が良いです。

sochi419 replied:
> dataはdatumの複数形なので、datasは英文法的にNGです。

そうなんですね、てっきりdatasという英単語があるものと思い込んでいました..

今回は、`datas`としていた箇所を
`data_stdin`, `data_argv`という名前にしてみました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,104 @@
+    word_count += input.split(/\s+/).size
+    bytesize += input.bytesize
+  end
+  [inputs.size, word_count, bytesize]
```
92〜102行目の処理と「似て非なるロジック」になっているため、全く同じテキストなのに、標準入力から渡したときと、ファイル名で指定したときで値が異なる、といった問題が起きるかもしれません。
同じロジックを使って値を算出するようにしてみましょう。

あと、なるべく配列の順番に意味を持たせない方が良いです。

[【初心者向け】順番に意味がある配列はハッシュに置き換えよう \- Qiita](https://qiita.com/jnchito/items/fb4ed2238ee3404b4403)

sochi419 replied:
> あと、なるべく配列の順番に意味を持たせない方が良いです。

配列ではなくハッシュを使った処理に変更しました。

> 同じロジックを使って値を算出するようにしてみましょう。

同じロジックを使って値を算出するよう修正しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,104 @@
+def output_stdin(opt, datas)
+  if !opt['l'] && !opt['w'] && !opt['c']
+    datas.each do |data|
+      output(data)
```
37-39行目のoutputと共通化したいですね。やりたいことは同じはずなので。

sochi419 replied:
37~39行目の部分を共通化しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,104 @@
+
+def build_data_for_argv(opt, argument)
+  str = File.read(argument)
+  output = if !opt['l'] && !opt['w'] && !opt['c']
```
こちらを参照してください〜。

[Rubyではメソッド名と同じ名前のローカル変数を宣言しないようにしよう \- Qiita](https://qiita.com/jnchito/items/fb438aa0786bb3cbf919)

sochi419 replied:
メソッド名とローカル変数名が同じ名前にならないように修正しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,104 @@
+  output << count_words(str) if opt['w']
+  output << count_byte(argument) if opt['c']
+
+  output
```
build_data_for_stdin では常に3つの配列を返しますが、こっちはoptの内容によって要素数が変わりますね。
こっちも常に3つ返すようにした方が、出力処理を再利用しやすくなります。

sochi419 replied:
要素数が3つになる様に統一しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,104 @@
+  output
+end
+
+def output_argv(_opt, datas, file)
```
やろうとしてることはほとんど同じはずなので、output_stdinと処理をまとめたいです。
あと、_optは使ってない変数ですね。

これはもしかすると「rubocopの自動修正にお任せー！わーい、警告消えた！！」で終わらせてしまったせいなのでは？（エスパー）

盲目的にrubocopの自動修正に頼っているとツールなしでは綺麗なコードが書けないプログラマができあがります。
安易な自動修正に頼らず、自分の手と目で修正していくのも勉強の一つです。

sochi419 replied:
> やろうとしてることはほとんど同じはずなので、output_stdinと処理をまとめたいです。

出力用のメソッドを1つにまとめました。

> これはもしかすると「rubocopの自動修正にお任せー！わーい、警告消えた！！」で終わらせてしまったせいなのでは？（エスパー）

正解です!笑 😅
rubocopの自動修正を過信していました。
今後は自動修正に頼らないようにしたいと思います。

余計な変数`_opt`は削除しました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,104 @@
+  total_lines_words_bytes.transpose.map(&:sum)
+end
+
+def output_total(opt, files)
```
これも`output_stdin`とかとまとめたいですね

sochi419 replied:
先ほどのコメント同様、出力用のメソッドを1つに統一しました。

JunichiIto reviewed (commented):
コメントしました！

sochi419 committed:
I have fixed the pointed out points.
・Combined common processing into one method.
・Change variable name

sochi419 commented:
指摘点について、修正をしました !
修正点については各コメントに記載しています。

よろしくお願いします🙏

sochi419 commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,91 @@
+    end
+    data_total = build_data_for_total(ARGV)
+    output(opt, data_total, nil, ARGV.size) if ARGV.size > 1
+  end
```
少し自身が無いのが、
出力用のメソッドを`output`に統一した結果、引数を4つ渡す必要があります。
ただ、場面によって使わない引数があるので、その部分を`nil`と置きました。

使わない引数の部分をnilと置くのは、やっていいのでしょうか？
（もう少しいい方法がありそうな気がする..🤔）

sochi419 commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,91 @@
+
+def output(opt, data, file, total_number_of_argv)
+  output_calculate_result(data, opt)
+  output_filename_or_total(file, total_number_of_argv)
```
出力用のoutputメソッドに、出力処理を全て詰め込んでいたのですが、
rubocopに複雑すぎると指摘を受けたので、以下の2つのメソッドに切り分けました。
・計算結果の出力
・ファイル名、totalの出力

sochi419 commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,91 @@
+  end
+end
+
+def output_filename_or_total(file, total_number_of_argv)
```
引数の総数という意味で、`total_number_of_argv`と命名しています。

sochi419 commented:
指摘点について、修正をしました !
修正点については各コメントに記載しています。

ちょっと自分の中で自身が無いなという部分があるので、
pushしたソースコードの中にコメントを加えています。

以上、確認よろしくお願いします🙏

JunichiIto reviewed (commented):
動画でレビューしたので別途リンクを送ります〜。

sochi419 committed:
Here are the main fixes:
・Change data structure
・Change of calculation process of total

sochi419 commented:
　動画にてレビューありがとうございます!
`動画` + `音声`だと、文章よりも説明の理解が容易にできるので、とても助かりました🙌

　動画にて指摘いただいた点をもとに、コードを修正しました。
変更点については、PR上のコードの中にコメントを加えました! 

確認よろしくお願いします🙏

sochi419 commented on 06.wc/wc.rb:
```diff
@@ -3,77 +3,52 @@
 
 def main
-  opt = ARGV.getopts('l', 'w', 'c')
+  options = ARGV.getopts('l', 'w', 'c')
```
`opt`という変数名をよりわかりやすくするために、
`options`という変数名に変更しました。

sochi419 commented on 06.wc/wc.rb:
```diff
@@ -3,77 +3,52 @@
-    output(opt, data_stdin, nil, nil)
+    output(build_data_for_stdin, options)
   else
+    total = { line: 0, word: 0, byte: 0, file: 'total' }
```
前回は出力処理の時と、合計値（total）を計算する時で、別々にループを行う処理をしていました。
ループが1度だけになるよう、出力と同時に`合計値(total)`を計算する処理に変更しました。

sochi419 commented on 06.wc/wc.rb:
```diff
@@ -3,77 +3,52 @@
-  end
-  { line: count_line_total, word: count_word_total, byte: count_byte_total }
+def get_calculate_result(str, file)
+  { line: count_line(str), word: count_words(str), byte: count_byte(str), file: file }
```
`build_data_for_stdin`と`build_data_for_file(file)`のデータ構造が同じだった為、
`get_calculate_result(str, file)`というメソッドを用意しました。

また、データ構造が`行数`、`単語数`、`バイト数`だったものに`file名`を追加しました。

sochi419 commented on 06.wc/wc.rb:
```diff
@@ -3,77 +3,52 @@
-  output_filename_or_total(file, total_number_of_argv)
-end
+def output(data, option)
+  print adjust_space(data[:line]) if option['l'] || lack_option(option)
```
前回の出力の処理
````
if !opt['l'] && !opt['w'] && !opt['c']
    print data[:line].to_s.rjust(8, ' ')
    print data[:word].to_s.rjust(8, ' ')
    print data[:byte].to_s.rjust(8, ' ')
  else
    print data[:line].to_s.rjust(8, ' ') if opt['l']
    print data[:word].to_s.rjust(8, ' ') if opt['w']
    print data[:byte].to_s.rjust(8, ' ') if opt['c']
````
は冗長なので、条件分岐の処理を変更しました。

JunichiIto replied:
lack_optionも不必要に何度も呼び出さない方がいいかもしれません

sochi419 replied:
`no_option = lack_option(option)`と定義して、
`no_option`を使い回す処理に変更しました！

sochi419 commented:
動画にてレビューありがとうございます!
アドバイスいただいた箇所を修正したところ、
かなり見やすくなった（DRYな）コードになったと思います！

指摘いただいた箇所については、コード上にコメントをしていますので、
確認よろしくお願いします🙏

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,66 @@
+  else
+    total = { line: 0, word: 0, byte: 0, file: 'total' }
+    ARGV.each do |file|
+      total[:line] += build_data_for_file(file)[:line].to_i
```
to_iは必要ですか？
あと、`build_data_for_file`を何度も呼び出すのは効率が悪いですね。

```ruby
def build_data_for_file(file)
  sleep 1
  # ...
end
```

のようにしてみると必要以上に時間がかかるのがわかると思いますー。

sochi419 replied:
`.to_i`は不要でした。

`calculate_result = build_data_for_file(file)`と変数を用意して、その変数を使い回す処理に変えました！

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,66 @@
+end
+
+def build_data_for_stdin
+  inputs = $stdin.readlines.join
```
複数形にすると配列っぽく見えるのでここでは不適かも。
文字列であることが伝わるような名前がいいですね。

sochi419 replied:
中身はファイル名が連結された文字列なので、変数名を`combined_filename`としました。

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,66 @@
+  get_calculate_result(str, file)
+end
+
+def get_calculate_result(str, file)
```
`build_data_for_stdin`、`build_data_for_file`と来てるのでここもbuildにした方が一貫性があるかも。
命名はちょっと悩みますが、文字列を受け取るので`build_data_for_text`とかどうでしょう？

sochi419 replied:
確かに`build_〇〇`ときているので、`build`で統一した方が一貫性が良さそうですね〜
`build_data_for_text`に変更しました！

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,66 @@
+  print adjust_space(data[:word]) if option['w'] || lack_option(option)
+  print adjust_space(data[:byte]) if option['c'] || lack_option(option)
+
+  print(' ')
+  print data[:file]
```
```suggestion
  print " #{data[:file]}"
```

好みの問題かもしれませんが、僕だったらこう書きますかねー。

sochi419 replied:
一つにまとめた方が、行数も減らせてコンパクトになると思います。
`print " #{data[:file]}"`にしました！

JunichiIto reviewed (commented):
コメントしました！

sochi419 committed:
The following points have been fixed.
・ Modified the process of replacing the method with a variable and reusing the variable.
・ Changed the name of the method.

sochi419 commented:
修正しました！

同じメソッドを使い回して、何度もメソッドを読んでしまう傾向がある様です。
今後は同じメソッドを使い回す場合は、「一度変数に置き換えてから使い回す」ようにしたいと思います！

以上、確認よろしくお願いします🙏

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -10,36 +10,37 @@ def main
+      total[:line] += calculate_result[:line]
+      total[:word] += calculate_result[:word]
+      total[:byte] += calculate_result[:byte]
       output(build_data_for_file(file), options)
```
ここにもまだ残ってまっせ〜！

sochi419 replied:
すみません、見落としていました...

この部分も同様に修正しました！

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -10,36 +10,37 @@ def main
-      total[:line] += build_data_for_file(file)[:line].to_i
-      total[:word] += build_data_for_file(file)[:word].to_i
-      total[:byte] += build_data_for_file(file)[:byte].to_i
+      calculate_result = build_data_for_file(file)
```
左辺と右辺の名前が全然違うので、どちらかの命名がまずってる可能性がありますね。

sochi419 replied:
ここは命名が難しかったです。
`build_data_for_file(file)`の出力が、
````
{ line: count_line(str), word: count_words(str), byte: count_byte(str), file: file }
````
なので、出力結果という意味で`calculate_result`としていました。

こちらに関しては、シンプルに
````
line_word_byte
````
としてみました。（厳密には、ハッシュの中にfile名の情報も入っていますが、冗長なので`line_word_byte_file`とはしませんでした。）

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -10,36 +10,37 @@ def main
 def build_data_for_stdin
-  inputs = $stdin.readlines.join
-  get_calculate_result(inputs, '')
+  combined_filename = $stdin.readlines.join
```
`$stdin.readlines.join`で返ってくるのはファイル名でしたっけ？そうとも限らないような🤔

sochi419 replied:
ファイル名しか渡せないものだと、勘違いしていました😅
試しにファイル名以外も標準入力で渡したところ、ちゃんと出力結果が返ってきました。

こちらに関しては、組み合わせた標準入力という意味で、
`joined_stndard_inputs`
と命名しました！

JunichiIto commented on 06.wc/wc.rb:
```diff
@@ -10,36 +10,37 @@ def main
-  print adjust_space(data[:line]) if option['l'] || lack_option(option)
-  print adjust_space(data[:word]) if option['w'] || lack_option(option)
-  print adjust_space(data[:byte]) if option['c'] || lack_option(option)
+  no_option = lack_option(option)
```
左辺と右辺の名前が微妙に違って気持ち悪いパターンですね（`person = find_user`みたいな形）
そもそもlack_optionは1回しか使われないのでメソッド化しない、という手もあります。
```suggestion
  no_option = !option['l'] && !option['w'] && !option['c']
```

sochi419 replied:
こちらはrubocopで、下記の様にメソッド内の複雑度が高いとエラーが出たため、
あえてメソッドを切り分けたという経緯です。
````
wc.rb:38:1: C: Metrics/PerceivedComplexity: Perceived complexity for output is too high. [9/8]
def output(data, option) ...
````

こちらに関しては、メソッドを切り分けない + rubocopからもエラーが出ない様な処理に変更しました！
````
def output(data, option)
  no_option = (option == { 'l' => false, 'w' => false, 'c' => false })　# trueもしくはfalseを返す
  print adjust_space(data[:line]) if option['l'] || no_option
  print adjust_space(data[:word]) if option['w'] || no_option
  print adjust_space(data[:byte]) if option['c'] || no_option

  print " #{data[:file]}"
  puts
end
````

JunichiIto replied:
個人的には別にこれぐらいいいじゃん、って思いますが、これだからrubocopは・・・おっと失礼w
なるほど、じゃあ仕方ないですね。ただ、`no_option = (option == { 'l' => false, 'w' => false, 'c' => false })`もちょっと可読性が悪い気がします。

`option.values`でハッシュの値を全部集めて、それから`none?`メソッドで真の値が1件でもあるかどうかをチェックする方法もあります。

```

irb(main):001:0> a=[false, false, true]
=> [false, false, true]
irb(main):002:0> b=[false, false, false] 
=> [false, false, false]
irb(main):003:0> c=[]
=> []
irb(main):007:0> a.none?
=> false
irb(main):008:0> b.none?
=> true
irb(main):009:0> c.none?
=> true
```

これもややトリッキーなところはありますが、コードはシンプルになりそうです。

sochi419 replied:
僕もこの部分、なんか気持ち悪いな〜🤔と思っていました。
````
no_option = (option == { 'l' => false, 'w' => false, 'c' => false })
````
なるほど、`values`メソッドと`none?`メソッドを組み合わせることで、かなり見やすくなった気がします！
````
no_option = option.values.none?
````
どちらも僕の知らないメソッドだったので、勉強になりました✍️

JunichiIto reviewed (commented):
コメントしました！

sochi419 committed:
Changed the variable name.
Fixed output processing.# with '#' will be ignored, and an empty message aborts the commit.

sochi419 commented:
指摘点を修正しました！
確認よろしくお願いします🙏

sochi419 committed:
Fixed the output method to be more readable code.

sochi419 commented:
コメントしました！
確認よろしくお願いします🙏

JunichiIto reviewed (approved):
確認しました。これでOKです〜 🙆‍♂️
